### ISSUE TITLE
The forwarding delay of netmap bridge is very high

### ISSUE THREAD

**Issue by wanttokiss** [luigirizzo/netmap#799]
Hello, why is the forwarding delay of netmap bridge very high? How to deal with this problem?

**Comment by jhk098** [comment]
Have you enabled promiscuous mode and disabled all the offloadings, as suggested by `LINUX/README.md`?

**Comment by wanttokiss** [comment]
Yes, I have referred to this document, and the parameters are attached. However, using the test instrument, it is found that in the case of bridge, the performance is relatively high, but the delay is very large。
![Uploading image.png…]()

**Comment by jhk098** [comment]
I see no documents. What NIC/driver are you using?

**Comment by wanttokiss** [comment]
Sorry, my description is wrong. The test process is NIC --- > app --- > kernel. After processing the data, the app sends the packet back to the kernel. However, the app does not process any data and just forwards it. Why is the delay so high.
64 bytes from 33.33.33.33: icmp_seq=144 ttl=128 time=6.54 ms
64 bytes from 33.33.33.33: icmp_seq=145 ttl=128 time=8.46 ms
64 bytes from 33.33.33.33: icmp_seq=146 ttl=128 time=6.68 ms
64 bytes from 33.33.33.33: icmp_seq=147 ttl=128 time=9.03 ms
64 bytes from 33.33.33.33: icmp_seq=148 ttl=128 time=6.99 ms
64 bytes from 33.33.33.33: icmp_seq=149 ttl=128 time=5.17 ms
64 bytes from 33.33.33.33: icmp_seq=150 ttl=128 time=7.14 ms
64 bytes from 33.33.33.33: icmp_seq=151 ttl=128 time=5.58 ms
64 bytes from 33.33.33.33: icmp_seq=152 ttl=128 time=3.48 ms
64 bytes from 33.33.33.33: icmp_seq=153 ttl=128 time=5.36 ms
64 bytes from 33.33.33.33: icmp_seq=154 ttl=128 time=8.47 ms
64 bytes from 33.33.33.33: icmp_seq=155 ttl=128 time=6.33 ms
64 bytes from 33.33.33.33: icmp_seq=156 ttl=128 time=8.66 ms
64 bytes from 33.33.33.33: icmp_seq=157 ttl=128 time=6.73 ms
64 bytes from 33.33.33.33: icmp_seq=158 ttl=128 time=6.25 ms

**Comment by giuseppelettieri** [comment]
netmap does not look at your ring unless you issue an `ioctl()` or `poll()`. If you care about latency, you need to limit your batching between two syscalls.
